<?xml version="1.0"?>
<launch>

    <!-- get the arguments from  -->
    <arg name="runs" default="runs.yaml" doc="Arrangements to Cycle Around" />
    <arg name="debug" default="False" doc="Whether debug is on" />
    <arg name="suffix" default="0" doc="Suffix to add to nodes" />
    <arg name="tag_topic" default="/tag_detections" doc="April tag topic" />
    <arg name="arrange_topic" default="/desired_arrange" doc="Arragement setpoint topic" />
    <arg name="rate" default="30" doc="Rate (in Hz) to run the controller (this should roughly match your camera's framerate)" />


    <!-- Generate helper arguments -->
    <arg name="_suffix_helper"     if="$(arg suffix)" value="_$(arg suffix)"  />
    <arg name="_suffix_helper" unless="$(arg suffix)" value=""  />

    <!-- Start the main serial sender -->
    <node name="cycle_node$(arg _suffix_helper)" pkg="tag2arrange" type="cycle_node.py" respawn="false"
        output="screen">
        <rosparam file="$(find tag2arrange)/config/cycle_test/$(arg runs)" command="load"/>
        <param name="DEBUG" type="bool" value="$(arg debug)"/>
        <param name="tag_topic" type="str" value="$(arg tag_topic)"/>
        <param name="arrange_topic" type="str" value="$(arg arrange_topic)"/>
        <param name="controller_rate" type="double" value="$(arg rate)"/>        
    </node>

</launch>